server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,468 @@
+            # move to backed-up-S3
+            _rename_tb_link(tb, os.path.join(
+                controller_path, "BACKED-UP-S3"), logger)
+        elif local_backup_result == Status.LOCAL_FAIL \
+            and s3_backup_result == Status.S3_FAIL:
+            # move tp backed-up-failed
+            _rename_tb_link(tb, os.path.join(
+                controller_path, "BACKED-UP-FAILED"), logger)
+        else:
+            assert False, "Logoc bomb: cannot happen: local_backup_result = \
+                {}, s3_backup_result = {}".format(local_backup_result,                                           s3_backup_result)
+
+    return Results(ntotal=ntotal,
+                   nbackup_success=nbackup_success,
+                   nbackup_fail=nbackup_fail,
+                   ns3_success=ns3_success,
+                   ns3_fail=ns3_fail,
+                   nquaran=nquaran)
+
+
+def main(parsed):
+    ret_status = 0
+    if not parsed.cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+              " use --config <file> on the command line".format(_NAME_),
+              file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(parsed.cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 1
+
+    global _logger
+    _logger = get_pbench_logger(_NAME_, config)
+
+    # call the S3Config class
+    s3_obj = S3Config(config)
+
+    _logger.info('start-{}'.format(config.timestamp()))
+
+    # add a BACKUP field to the config object
+    config.BACKUP = config.conf.get("pbench-server", "pbench-backup-dir")
+    config.QDIR = config.get('pbench-server', 'pbench-quarantine-dir')
+
+    # call the sanity check function
+    sanity_check(s3_obj, config, _logger)
+
+    prog = os.path.basename(sys.argv[0])
+
+    # Initiate the backup
+    counts = backup_data(s3_obj, config, _logger)
+
+    result_string = ("Total processed: {}, "
+                     "Locally backed-ed succesfully: {}, "
+                     "Failed to locally backed-up: {}, "
+                     "Uploaded to S3 succesfully: {}, "
+                     "Failed to upload to S3: {}, Quarantine: {}"
+                    .format(counts.ntotal,
+                            counts.nbackup_success,
+                            counts.nbackup_fail,
+                            counts.ns3_success,
+                            counts.ns3_fail,
+                            counts.nquaran))
+
+    _logger.info(result_string)
+
+    with tempfile.NamedTemporaryFile(mode='w+t') as report:
+        # call prepare report function
+        prepare_report(report, config.timestamp(), prog, config, result_string)
+        report.seek(0)
+
+        es, idx_prefix = init_report_template(config, _logger)
+        # Call report-status
+        report_status(es, _logger, config.LOGSDIR,
+                      idx_prefix, _NAME_, config.timestamp(), "status", (pathlib.Path(report.name)))
```

Comment:
What does `pathlib.Path` do?

Reply:
since we already have an opened temporary file, as a `report` file descriptor. In `report_status` we cannot open it again, it will give a type error: `TypeError: expected str, bytes or os.PathLike object, not _TemporaryFileWrapper`. `pathlib.path` will provide the full path of the temporary file we have created.

Reply:
Doesn't `report.name` provide that?